fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -462,6 +477,21 @@ public CompletableFuture<Void> commitAsync() {
                 return AsyncUtil.DONE;
             }
         });
+        if (config.isReportConflictingKeys()) {
+            // Do this before the close (below).
+            commit = MoreAsyncUtil.composeWhenCompleteAndHandle(commit, (v, ex) -> {
+                if (ex instanceof CompletionException) {
+                    final Throwable cause = ex.getCause();
+                    if (cause instanceof FDBException) {
+                        final FDBException fdbException = (FDBException)cause;
+                        if (FDBError.fromCode(fdbException.getCode()) == FDBError.NOT_COMMITTED) {
+                            return readConflictingKeys();
```

Comment:
It would be nice to have a test of this. Something like a test that constructs two transactions with the appropriate options and then orchestrates a conflict and validates that the conflicting ranges read align